test/utils/update_resources.go

```diff
@@ -33,10 +34,25 @@ const (
 	waitRetryTImeout    = 5 * time.Minute
 )
 
+func scaleCondition(scaler kubectl.Scaler, namespace, name string, size uint, gr schema.GroupResource, err *error) wait.ConditionFunc {
+	return func() (bool, error) {
+		waitForScale := kubectl.NewRetryParams(updateRetryInterval, updateRetryTimeout)
+		waitForReplicas := kubectl.NewRetryParams(waitRetryInterval, waitRetryTImeout)
+		if *err = scaler.Scale(namespace, name, size, nil, waitForScale, waitForReplicas, gr); *err != nil {
```

Comment:
Did you read my comment I added yesterday? Did isn't correct.

This isn't what we want, because of couple reasons:
- in theory it can take much more than we want it to take (in terms of wall time)
- it is not correct - scaler.Scale underneath does two things:
(a) changes size of grouping object (e.g. deployment)
(b) waits until cluster state converges to that state

If we correctly updated object and then had error in second phase, we don't want to do (a) again.
If we timedout on waiting for reconciliation, that's a valid error, which we don't want to retry.

Please take a look into my comment and follow what I suggested there.